### PR TITLE
feat: Azurite Deployment and related configurations

### LINKED ISSUE #124: Deploy azurite into MXD
# Description
Create terraform configs for:

- Azurite Kubernetes Deployment
- Azurite Kubernetes Service
- Initial Script / Job to create buckets for Alice / Bob.

Related to eclipse-tractusx/sig-release#315

### PR DESCRIPTION
**Pull request by hemantxpatel**
## Description

- Terraform azurite module for azurite deployment and service.
- Call azurite module from `main.tf`. (In future we can call from connector module as well in case of one deployment per connector)
- Terraform seed job to create azure container and upload a sample document file for each connector.
- Update connector's config (blob template, vault seed with azure account key etc.).

Closes #124 
## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: mxd/modules/connector/values.yaml [hunk 1]
```diff
@@ -24,6 +24,10 @@
 participant:
   id: "test-participant"
 controlplane:
+#  livenessProbe:
```
Comment (paullatzelsperger) [inline mxd/modules/connector/values.yaml:27]: any reason you added this?

File: mxd/modules/connector/variables.tf [hunk 2]
```diff
@@ -62,4 +62,21 @@ variable "ssi-config" {
     oauth-clientsecret = ""
     oauth-secretalias  = ""
   }
-}
\ No newline at end of file
+}
+
+variable "azure-account-name" {
```
Comment (paullatzelsperger) [inline mxd/modules/connector/variables.tf:67]: could provide a default value, something like `"${var.humanreadablename}-account` or similar

File: mxd/modules/connector/variables.tf [hunk 3]
```diff
@@ -62,4 +62,21 @@ variable "ssi-config" {
...
+  description = "Azure Account Name for the connector"
+}
+
+variable "azure-account-key" {
+  description = "Azure Account Key for the connector"
```
Comment (paullatzelsperger) [inline mxd/modules/connector/variables.tf:72]: could provide a default value, eg. a random generated one

File: mxd/modules/connector/azurite-container.tf [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#  License for the specific language governing permissions and limitations
+#  under the License.
+#
+#  SPDX-License-Identifier: Apache-2.0
+#
+
+resource "kubernetes_job" "azurite-container" {
+
+  metadata {
```
Comment (paullatzelsperger) [inline mxd/modules/connector/azurite-container.tf:22]: here, i would add an explicit `dependsOn` entry, to make sure this job runs after the azurite container was deployed

File: mxd/modules/connector/azurite-container.tf [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    name = "${local.appName}-document"
+  }
+  data = {
+    (local.file-name) = file("./assets/${local.file-name}")
+  }
+}
+
+locals {
+  appName              = "${var.humanReadableName}-container"
```
Comment (paullatzelsperger) [inline mxd/modules/connector/azurite-container.tf:69]: that is not very expressive. something like `..-azurite-init` would be better

File: mxd/modules/connector/azurite-container.tf [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#  License for the specific language governing permissions and limitations
+#  under the License.
+#
+#  SPDX-License-Identifier: Apache-2.0
+#
+
+resource "kubernetes_job" "azurite-container" {
```
Comment (paullatzelsperger) [inline mxd/modules/connector/azurite-container.tf:20]: not for this PR, but we should keep seeding the data backends (Azurite) and the EDC data (via management API) consistent, meaning:
either we create one script to seed both connectors (`seed_data.tf`), or we create the seed script "per connector", like you did with Azurite. There is advantages to either variant.

**Review by paullatzelsperger — COMMENTED** [review]
looks good, with some minor comments

File: mxd/modules/connector/values.yaml (hunk 1 above)
Comment (hemantxpatel) [reply mxd/modules/connector/values.yaml:27]: When I do remote debug and put a breakpoint for a long time in our connector's data-plane / control-plane, kubernetes probes fail and it restarts the pod.
So I often need these configs to disable the probes. Just that, I had to copy and paste every time, now I have to just uncomment.

File: mxd/modules/connector/variables.tf (hunk 2 above)
Comment (hemantxpatel) [reply mxd/modules/connector/variables.tf:67]: Terraform doesn't support variable reference inside `variables.tf`. `terraform init` fails.
https://stackoverflow.com/questions/58841060/terraform-variables-within-variables

We do have default values for alice / bob accounts in top level variables.tf.

File: mxd/modules/connector/variables.tf (hunk 3 above)
Comment (hemantxpatel) [reply mxd/modules/connector/variables.tf:72]: We do have a default value for both Alice and Bob in top level variables.tf.
We have some limitations on using a random generated password.
Azure temp keys (Azure SAS) has been generated using the default account name / account key and added as static value in the code.
We can generate Azure SAS dynamically, using Azure CLI pod, but we can't share this key to the connector pod as a variable in vault postStart script (kind of Terraform limitation, I asked about it in reddit and they suggested to use ArgoCD, ArgoCD does support workflows where output of one pod can be shared with next pod in the workflow). Good thing is this SAS key doesn't change if parameters are same (Account Name / key / permissions etc), that's why I pre-generated it and added as static value.
Also I have added steps how to generate this SAS key manually in the tutorial PR (#157).

File: mxd/modules/connector/azurite-container.tf (hunk 4 above)
Comment (hemantxpatel) [reply mxd/modules/connector/azurite-container.tf:22]: Here `module.azurite` is not visible. Added explicit dependency in connector. So now, connectors depends on azurite module.

File: mxd/modules/connector/azurite-container.tf (hunk 5 above)
Comment (hemantxpatel) [reply mxd/modules/connector/azurite-container.tf]: Done, pod names will be `alice-azurite-init` and `bob-azurite-init`

File: mxd/modules/connector/values.yaml (hunk 1 above)
Comment (paullatzelsperger) [reply mxd/modules/connector/values.yaml:27]: we should not have commented code in there. i would prefer to have it active, and you can comment it out when debugging. pls uncomment.

File: mxd/modules/connector/variables.tf (hunk 2 above)
Comment (paullatzelsperger) [reply mxd/modules/connector/variables.tf:67]: ah right, forgot.

**Review by paullatzelsperger — APPROVED** [review]

**Review by paullatzelsperger — COMMENTED** [review]
all good, except the commented code. see inline.

File: mxd/modules/connector/values.yaml (hunk 1 above)
Comment (hemantxpatel) [reply mxd/modules/connector/values.yaml]: I enabled it to `true` (which is also the default value in helm chart) and while debugging, I can make it `false`.

**Review by paullatzelsperger — APPROVED** [review]

**Comment by paullatzelsperger** [comment]
@hemantxpatel can you rebase and resolve conflicts? then we're good to merge.

**Pull request merged**